superagentx_handlers/websitecrawler/amazon.py

```diff
@@ -0,0 +1,190 @@
+from superagentx.handler.base import BaseHandler
+from superagentx.handler.decorators import tool
+from superagentx.utils.helper import sync_to_async
+
+logger = logging.getLogger(__name__)
+
+
+class AmazonWebHandler(BaseHandler):
+
+    def __init__(
+            self,
+    ):
+        super().__init__()
+        self.deals_list = []
+        self.asession = AsyncHTMLSession()
+
+    async def _get_data(
+            self,
+            url: str
+    ):
+        headers = {
+            'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)'
+                          ' Chrome/98.0.4758.80 Safari/537.36',
+            'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive',
+            'accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,'
+                      'image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
+            'rtt': '200'
+        }
+        r = await self.asession.get(
+            url,
+            headers=headers
+        )
+        return BeautifulSoup(
+            r.html.html,
+            'html.parser'
+        )
+
+    def _get_deals(
+            self,
+            soup
+    ):
+        products = soup.find_all(
+            'div',
+            {
+                'data-component-type': 's-search-result'
+            }
+        )
+        if products:
+            for item in products:
+                title = item.find(
+                    'a',
+                    {
+                        'class': 'a-link-normal s-underline-text s-underline-link-text s-link-style a-text-normal'
+                    }
+                ).text.strip()
+                link = item.find(
+                    'a',
+                    {
+                        'class': 'a-link-normal s-underline-text s-underline-link-text s-link-style a-text-normal'
+                    }
+                ).get('href')
+                try:
+                    sale_prices = item.find_all('span', {'class': 'a-offscreen'})
+                    if sale_prices:
+                        sale_price = sale_prices[0].text.replace('$', '').replace(',', '').strip()
+                    else:
+                        sale_price = ''
+
+                    old_prices = item.find_all('span', {'class': 'a-offscreen'})
+                    if old_prices:
+                        old_price = old_prices[1].text.replace('$', '').replace(',', '').strip()
+                    else:
+                        old_price = ''
+                except:
+                    old_prices = item.find(
+                        'span',
+                        {
+                            'class': 'a-offscreen'
+                        }
+                    )
+                    old_price = float(
+                        old_prices.text.replace('£', '').replace(',', '').strip())
+                try:
+                    reviews = float(
+                        item.find(
```

Comment:
Use `sync_to_async` method from utils to make this as `async`

Reply:
fixed